docs/start.md

```diff
@@ -133,6 +133,11 @@ For more information on updating an existing service account:
 - Granting roles:
   [granting-roles-to-service-accounts](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts)
 
+SSH
+---
+Requires no provider credentials to test instances. SSH user, SSH private key can
```

Comment:
Lets also support a pointer to a private key file. Differentiating his in the code is a simple if condition.

Reply:
This is a key file. It is poorly named:

`--ssh-private-key PATH          SSH private key file for accessing instance.`

We can change the name for CLI but probably want to deprecate it then drop it in a major release:

`--ssh-private-key-file PATH          SSH private key file for accessing instance.`

I don't think it should be expected that anyone would pass a private key in command line so just providing the key file option seems good.

Reply:
Please change according to the other Provider class changes

Reply:
> Please change according to the other Provider class changes

Rebased against master.